packages/web3-providers-ws/src/index.js

```diff
@@ -171,6 +173,18 @@ WebsocketProvider.prototype._addResponseCallback = function(payload, callback) {
 
     this.responseCallbacks[id] = callback;
     this.responseCallbacks[id].method = method;
```

Comment:
Doesnt look like youre clearing the custom timeout when the response came. also keep in mind that certain calls, like for past logs etc take very long to answer. Sometimes 10+ seconds

Reply:
@frozeman thanks for the comment!

Actually, clearing the timeout is not needed and would add complexity IMHO. Once the actual response comes in, the function at `this.responseCallbacks[id]` will be executed and deleted from the dictionary in L79. As long as ids are unique, there will be no issues. And, in fact, if ids are not unique, many other things would break.

Re calls with long answer time, it would be up to the user to select a proper timeout. This PR does not enforce nor provides any default. Having said that, is a very good observation. Do you think changing the README so the example is i.e. 30s instead of 5s would clear any doubts it may generate?

Reply:
default of 30s would be better yes

Reply:
and i would simply make the sample as such: `var ws = new Web3WsProvider('ws://localhost:8546', {timeout: 30000});`